metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java

```diff
@@ -638,4 +654,23 @@ protected static AspectSpec lookupAspectSpec(EntitySpec entitySpec, String aspec
             .findFirst()
             .get();
   }
+
+  protected static Urn validatedUrn(String urn) throws InvalidUrnException {
+    try {
+      return Urn.createFromString(urn);
+    } catch (URISyntaxException e) {
+      throw new InvalidUrnException(urn, "Invalid urn!");
+    }
+  }
+
+  public static class InvalidUrnException extends URISyntaxException {
+    public InvalidUrnException(String input, String reason) {
+      super(input, reason);
+    }
+  }
+
+  @ExceptionHandler(InvalidUrnException.class)
```

Comment:
Should this be applied globally? Might make more sense to have it in the GlobalControllerExceptionHandler